### PR TITLE
[WEBEDIT INCOMING]Allows any atom that is not an effect to increase supermatter matter power on consumption.

### PR DESCRIPTION
**Pull request by Pickle-Coding**
If the consumed_object does not pass the isitem check, then it increases the matter_increase by half of consumed_object.max_integrity, capped at 1000.
# WEBEDIT INFORMATION
This is indeed a webedit, but I tested the changes on local.

## About The Pull Request
If the atom is not an item, the matter power increase will be half of the object's max integrity, up to a maximum of 1GeV (in case some object has an absurd amount of integrity).
## Why It's Good For The Game
When a physical object gets consumed by the supermatter, people expect it to do what it does for items, to give it trickled power over time. It's confusing when it doesn't increase the power.
## Changelog
:cl:
fix: Objects that are not items can increase supermatter power on consumption.
/:cl:

### CODE REVIEW

**Comment by TheVekter** [comment]
Can we do something about the kind of lag these set-ups cause before actually making it more attractive to set them up? Someone did a pill SM on Manuel last round and looking at it tanked the hell out of my response time.

**Comment by TemporalOroboros** [comment]
Most of the lag is probably from the irradiation/hallucination pulses. It should be pretty trivial to buffer them until the next tick or something like that.

**Comment by Pickle-Coding** [comment]
> Most of the lag is probably from the irradiation/hallucination pulses. It should be pretty trivial to buffer them until the next tick or something like that.

Radiation pulses check tick. Not sure about hallucination pulses, but I could probably merge hallucination pulses with radiation pulses for processing in a different PR, as I imagine the hallucination pulses are a unique form of radiation.

**Comment by Pickle-Coding** [comment]
> Can we do something about the kind of lag these set-ups cause before actually making it more attractive to set them up? Someone did a pill SM on Manuel last round and looking at it tanked the hell out of my response time.

Is there a case where someone could easily produce a bunch of atoms that are not items?

File: code/datums/components/supermatter_crystal.dm [hunk 1]
```diff
@@ -259,10 +259,13 @@
...
-		var/obj/item/consumed_item = consumed_object
-		object_size = consumed_item.w_class
-		matter_increase += 70 * object_size
+	if(!iseffect(consumed_object))
+		if(isitem(consumed_object))
+			var/obj/item/consumed_item = consumed_object
+			object_size = consumed_item.w_class
+			matter_increase += 70 * object_size
+		else
+			matter_increase += min(0.5 * consumed_object.max_integrity, 4000)
```
Comment (LemonInTheDark) [inline code/datums/components/supermatter_crystal.dm:268]: please max this lower

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]
What are you imagining this to cover? also does this mean mobs double count

**Comment by Pickle-Coding** [comment]
> What are you imagining this to cover? also does this mean mobs double count

Like crates, loose girders, vehicles, and other stuff. Are mobs a subtype of atom?

File: code/datums/components/supermatter_crystal.dm [hunk 2]
```diff
@@ -259,10 +259,13 @@
...
-	if(!iseffect(consumed_object) && isitem(consumed_object))
-		var/obj/item/consumed_item = consumed_object
-		object_size = consumed_item.w_class
-		matter_increase += 70 * object_size
+	if(!iseffect(consumed_object))
+		if(isitem(consumed_object))
+			var/obj/item/consumed_item = consumed_object
+			object_size = consumed_item.w_class
+			matter_increase += 70 * object_size
+		else
```
Comment (Pickle-Coding) [inline code/datums/components/supermatter_crystal.dm:267]:
```suggestion
		else if(!isliving(consumed_object))
```

File: code/datums/components/supermatter_crystal.dm (hunk 1 above)
Comment (Pickle-Coding) [inline code/datums/components/supermatter_crystal.dm:268]:
```suggestion
			matter_increase += min(0.5 * consumed_object.max_integrity, 1000)
```

**Comment by LemonInTheDark** [comment]
Anything that exists in the world is an atom

File: code/datums/components/supermatter_crystal.dm [hunk 3]
```diff
@@ -259,10 +259,13 @@
 				message_admins("[atom_source] has consumed [consumed_object], [suspicion] [ADMIN_JMP(atom_source)].")
 			atom_source.investigate_log("has consumed [consumed_object] - [suspicion].", INVESTIGATE_ENGINE)
 		qdel(consumed_object)
-	if(!iseffect(consumed_object) && isitem(consumed_object))
-		var/obj/item/consumed_item = consumed_object
-		object_size = consumed_item.w_class
-		matter_increase += 70 * object_size
+	if(!iseffect(consumed_object))
```
Comment (LemonInTheDark) [inline code/datums/components/supermatter_crystal.dm:262]: just do the check up here

Comment (Pickle-Coding) [reply code/datums/components/supermatter_crystal.dm:262]: Oh.

File: code/datums/components/supermatter_crystal.dm [hunk 4]
```diff
@@ -259,10 +259,13 @@
...
+		if(isitem(consumed_object))
+			var/obj/item/consumed_item = consumed_object
+			object_size = consumed_item.w_class
+			matter_increase += 70 * object_size
+		else if(!isliving(consumed_object))
+			matter_increase += min(0.5 * consumed_object.max_integrity, 1000)
```
Comment (Pickle-Coding) [inline code/datums/components/supermatter_crystal.dm:268]:
```suggestion
	if(!iseffect(consumed_object) && !isliving(consumed_object))
		if(isitem(consumed_object))
			var/obj/item/consumed_item = consumed_object
			object_size = consumed_item.w_class
			matter_increase += 70 * object_size
		else
			matter_increase += min(0.5 * consumed_object.max_integrity, 1000)
```

**Review by Pickle-Coding — COMMENTED** [review]
lol

**Comment by LemonInTheDark** [comment]
If this dusts ghosts I will bully you so hard

**Pull request merged**